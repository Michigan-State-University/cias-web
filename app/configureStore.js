/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createSagaMiddleware from 'redux-saga';

import utilsHistory from 'utils/history';
import { initialState as authReducerInitialState } from 'global/reducers/auth/reducer';

import { loadState } from 'utils/persist';
import { createInjectorsEnhancer } from 'redux-injectors';
import createReducer from './reducers';

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        serialize: {
          options: {
            map: true,
          },
        },
      });

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    // if (window.__SAGA_MONITOR_EXTENSION__)
    //   reduxSagaMonitorOptions = {
    //     sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
    //   };
    /* eslint-enable */
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const runSaga = sagaMiddleware.run;

  const enhancers = [
    applyMiddleware(...middlewares),
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(store.createReducer(store.injectedReducers));
    });
  }

  return store;
}
const { user } = loadState() ?? {};

const initialStore = { auth: { ...authReducerInitialState, user } };

export const store = configureStore(initialStore, utilsHistory);
