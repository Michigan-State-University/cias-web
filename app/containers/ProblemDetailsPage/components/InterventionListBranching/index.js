/**
 *
 * InterventionListBranching
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import Row from 'components/Row';
import Column from 'components/Column';

import Switch from 'components/Switch';
import Text from 'components/Text';
import BranchingLayout from 'containers/BranchingLayout';

import messages from './messages';

const initFormula = {
  payload: '',
  patterns: [{ match: '', target: { id: '', type: 'Intervention' } }],
};

function InterventionListBranching({
  intl: { formatMessage },
  nextInterventionName,
  branching,
  handleBranching,
}) {
  const [formula, setFormula] = useState(initFormula);
  return (
    <>
      <Row py={18} px={62} align="between" justify="between">
        <Column xs={12}>
          <Row align="center" width="100%">
            {(branching || nextInterventionName) && (
              <>
                <Text fontSize={13}>{formatMessage(messages.nextSession)}</Text>
                <Text ml={10} fontSize={13} fontWeight="bold">
                  {branching
                    ? formatMessage(messages.formula)
                    : nextInterventionName}
                </Text>
              </>
            )}
          </Row>
        </Column>
        <Column xs={3}>
          <Row justify="end" align="center" width="100%">
            <Text>{formatMessage(messages.useFormula)}</Text>
            <Switch ml={10} checked={branching} onToggle={handleBranching} />
          </Row>
        </Column>
      </Row>
      {branching && (
        <Row mx={62} py={20}>
          <Column>
            <BranchingLayout
              onUpdateCase={() => null}
              onRemoveCase={() => null}
              onFormulaUpdate={() => null}
              onAddCase={() =>
                setFormula({
                  ...formula,
                  patterns: [...formula.patterns, initFormula.patterns[0]],
                })
              }
              displayPatternTargetText={() => 'test-1'}
              formatMessage={formatMessage}
              formula={formula}
              id="test-1"
              problemBranching
            />
          </Column>
        </Row>
      )}
    </>
  );
}

InterventionListBranching.propTypes = {
  intl: PropTypes.object,
  nextInterventionName: PropTypes.string,
  branching: PropTypes.bool,
  handleBranching: PropTypes.func,
};

export default injectIntl(InterventionListBranching);
