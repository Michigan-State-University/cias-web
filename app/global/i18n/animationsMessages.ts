import { defineMessages } from 'react-intl';

import { NarratorAnimation } from 'models/Narrator';

export const scope = 'app.global.Animations';

export default defineMessages<NarratorAnimation>({
  uncertain: {
    id: `${scope}.uncertain`,
    defaultMessage: 'Uncertain',
  },
  wave: {
    id: `${scope}.wave`,
    defaultMessage: 'Wave',
  },
  greet: {
    id: `${scope}.greet`,
    defaultMessage: 'Greet',
  },
  otohLeftHand: {
    id: `${scope}.otohLeftHand`,
    defaultMessage: 'On the other hand (left)',
  },
  otohRightHand: {
    id: `${scope}.otohRightHand`,
    defaultMessage: 'On the other hand (right)',
  },
  pointRight: {
    id: `${scope}.pointRight`,
    defaultMessage: 'Point right',
  },
  pointLeft: {
    id: `${scope}.pointLeft`,
    defaultMessage: 'Point left',
  },
  pointUp: {
    id: `${scope}.pointUp`,
    defaultMessage: 'Point up',
  },
  pointDown: {
    id: `${scope}.pointDown`,
    defaultMessage: 'Point down',
  },
  pointLeftUp: {
    id: `${scope}.pointLeftUp`,
    defaultMessage: 'Point left up',
  },
  pointRightUp: {
    id: `${scope}.pointRightUp`,
    defaultMessage: 'Point right up',
  },
  explain: {
    id: `${scope}.explain`,
    defaultMessage: 'Explain',
  },
  rest: {
    id: `${scope}.rest`,
    defaultMessage: 'Rest',
  },
  restShoulderRubbing: {
    id: `${scope}.restShoulderRubbing`,
    defaultMessage: 'Rest shoulder rubbing',
  },
  restStretching: {
    id: `${scope}.restStretching`,
    defaultMessage: 'Rest stretching',
  },
  restWeightShift: {
    id: `${scope}.restWeightShift`,
    defaultMessage: 'Rest weight shift',
  },
  read: {
    id: `${scope}.read`,
    defaultMessage: 'Read',
  },
  reading: {
    id: `${scope}.reading`,
    defaultMessage: 'Reading',
  },
  confused: {
    id: `${scope}.confused`,
    defaultMessage: 'Confused',
  },
  doMagic: {
    id: `${scope}.doMagic`,
    defaultMessage: 'Do magic',
  },
  getAttention: {
    id: `${scope}.getAttention`,
    defaultMessage: 'Get attention',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
  suggest: {
    id: `${scope}.suggest`,
    defaultMessage: 'Suggest',
  },
  surprised: {
    id: `${scope}.surprised`,
    defaultMessage: 'Surprised',
  },
  congratulate: {
    id: `${scope}.congratulate`,
    defaultMessage: 'Congratulate',
  },
  flyIn: {
    id: `${scope}.flyIn`,
    defaultMessage: 'Fly in',
  },
  flyOut: {
    id: `${scope}.flyOut`,
    defaultMessage: 'Fly out',
  },
  standStill: {
    id: `${scope}.standStill`,
    defaultMessage: 'Stand still',
  },
  process: {
    id: `${scope}.process`,
    defaultMessage: 'Process',
  },
  listen: {
    id: `${scope}.listen`,
    defaultMessage: 'Listen',
  },
  think: {
    id: `${scope}.think`,
    defaultMessage: 'Think',
  },
  write: {
    id: `${scope}.write`,
    defaultMessage: 'Write',
  },
  idle: {
    id: `${scope}.idle`,
    defaultMessage: 'Idle',
  },
  turnPage: {
    id: `${scope}.turnPage`,
    defaultMessage: 'Turn page',
  },
  openBook: {
    id: `${scope}.openBook`,
    defaultMessage: 'Open book',
  },
  closeBook: {
    id: `${scope}.closeBook`,
    defaultMessage: 'Close book',
  },
  announce: {
    id: `${scope}.announce`,
    defaultMessage: 'Announce',
  },
  blink: {
    id: `${scope}.blink`,
    defaultMessage: 'Blink',
  },
  acknowledge: {
    id: `${scope}.acknowledge`,
    defaultMessage: 'Acknowledge',
  },
  acknowledgeThoughtful: {
    id: `${scope}.acknowledgeThoughtful`,
    defaultMessage: 'Acknowledge thoughtful',
  },
  browsUp: {
    id: `${scope}.browsUp`,
    defaultMessage: 'Brows up',
  },
  decline: {
    id: `${scope}.decline`,
    defaultMessage: 'Decline',
  },
  declineAnnoyed: {
    id: `${scope}.declineAnnoyed`,
    defaultMessage: 'Decline annoyed',
  },
  declineThoughtful: {
    id: `${scope}.declineThoughtful`,
    defaultMessage: 'Decline thoughtful',
  },
  pleased: {
    id: `${scope}.pleased`,
    defaultMessage: 'Pleased',
  },
  sad: {
    id: `${scope}.sad`,
    defaultMessage: 'Sad',
  },
  hearLeftEar: {
    id: `${scope}.hearLeftEar`,
    defaultMessage: 'Hear left ear',
  },
  hearRightEar: {
    id: `${scope}.hearRightEar`,
    defaultMessage: 'Hear right ear',
  },
  hearBothEars: {
    id: `${scope}.hearBothEars`,
    defaultMessage: 'Hear both ears',
  },
  yawn: {
    id: `${scope}.yawn`,
    defaultMessage: 'Yawn',
  },
  eatCracker: {
    id: `${scope}.eatCracker`,
    defaultMessage: 'Eat cracker',
  },
  wearSunglasses: {
    id: `${scope}.wearSunglasses`,
    defaultMessage: 'Wear sunglasses',
  },
  lookDown: {
    id: `${scope}.lookDown`,
    defaultMessage: 'Look down',
  },
  lookDownAndBlink: {
    id: `${scope}.lookDownAndBlink`,
    defaultMessage: 'Look down and blink',
  },
  glanceUp: {
    id: `${scope}.glanceUp`,
    defaultMessage: 'Glance up',
  },
  glanceDown: {
    id: `${scope}.glanceDown`,
    defaultMessage: 'Glance down',
  },
  glanceLeft: {
    id: `${scope}.glanceLeft`,
    defaultMessage: 'Glance left',
  },
  glanceRight: {
    id: `${scope}.glanceRight`,
    defaultMessage: 'Glance right',
  },
  moveLeft: {
    id: `${scope}.moveLeft`,
    defaultMessage: 'Move left',
  },
  moveRight: {
    id: `${scope}.moveRight`,
    defaultMessage: 'Move right',
  },
});
