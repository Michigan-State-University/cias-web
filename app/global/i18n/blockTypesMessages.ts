import { defineMessages } from 'react-intl';

export const scope = 'app.global.BlockTypes';

export default defineMessages({
  BodyAnimation: {
    id: `${scope}.BodyAnimation`,
    defaultMessage: 'Body Animation',
  },
  Speech: {
    id: `${scope}.Speech`,
    defaultMessage: 'Speech',
  },
  Reflection: {
    id: `${scope}.Reflection`,
    defaultMessage: 'Speech',
  },
  ReflectionFormula: {
    id: `${scope}.ReflectionFormula`,
    defaultMessage: 'Speech',
  },
  HeadAnimation: {
    id: `${scope}.HeadAnimation`,
    defaultMessage: 'Head Animation',
  },
  ReadQuestion: {
    id: `${scope}.ReadQuestion`,
    defaultMessage: 'Read question',
  },
  Pause: {
    id: `${scope}.Pause`,
    defaultMessage: 'Pause',
  },
  Feedback: {
    id: `${scope}.Feedback`,
    defaultMessage: 'Show Spectrum',
  },
});
