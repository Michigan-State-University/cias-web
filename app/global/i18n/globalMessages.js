import { defineMessages } from 'react-intl';

export const scope = 'app.GlobalMessages';

export default defineMessages({
  variables: {
    variableNamePlaceholder: {
      id: `${scope}.variableNamePlaceholder`,
      defaultMessage: 'Variable name...',
    },
    variableScorePlaceholder: {
      id: `${scope}.variableScorePlaceholder`,
      defaultMessage: 'Score',
    },
    value: {
      id: `${scope}.value`,
      defaultMessage: 'Value:',
    },
    emptyBadge: {
      id: `${scope}.emptyBadge`,
      defaultMessage: 'No assigned value',
    },
  },
  blockTypes: {
    BodyAnimation: {
      id: `${scope}.BodyAnimation`,
      defaultMessage: 'Body Animation',
    },
    Speech: {
      id: `${scope}.Speech`,
      defaultMessage: 'Speech',
    },
    Reflection: {
      id: `${scope}.Reflection`,
      defaultMessage: 'Speech',
    },
    HeadAnimation: {
      id: `${scope}.HeadAnimation`,
      defaultMessage: 'Head Animation',
    },
    ReadQuestion: {
      id: `${scope}.ReadQuestion`,
      defaultMessage: 'Read question',
    },
  },
  questionTypes: {
    'Question::Single': {
      id: `${scope}.Question::Single`,
      defaultMessage: 'Single answer screen',
    },
    'Question::Multiple': {
      id: `${scope}.Question::Multiple`,
      defaultMessage: 'Multi answer screen',
    },
    'Question::TextBox': {
      id: `${scope}.Question::TextBox`,
      defaultMessage: 'Textbox screen',
    },
    'Question::Number': {
      id: `${scope}.Question::Number`,
      defaultMessage: 'Number screen',
    },
    'Question::Grid': {
      id: `${scope}.Question::Grid`,
      defaultMessage: 'Grid screen',
    },
    'Question::AnalogueScale': {
      id: `${scope}.Question::AnalogueScale`,
      defaultMessage: 'Visual Analogue Scale screen',
    },
    'Question::Information': {
      id: `${scope}.Question::Information`,
      defaultMessage: 'Information screen',
    },
    'Question::Url': {
      id: `${scope}.Question::Url`,
      defaultMessage: 'Url screen',
    },
    'Question::FollowUpContact': {
      id: `${scope}.Question::FollowUpContact`,
      defaultMessage: 'FollowUpContact screen',
    },
  },
  statuses: {
    draft: {
      id: `${scope}.draft`,
      defaultMessage: 'Draft',
    },
    published: {
      id: `${scope}.published`,
      defaultMessage: 'Published',
    },
    closed: {
      id: `${scope}.closed`,
      defaultMessage: 'Closed',
    },
    archived: {
      id: `${scope}.archived`,
      defaultMessage: 'Archived',
    },
  },
  createProblemError: {
    id: `${scope}.createProblemError`,
    defaultMessage: `Couldn't create an intervention`,
  },
  editProblemError: {
    id: `${scope}.editProblemError`,
    defaultMessage: `Couldn't edit an intervention`,
  },
});
