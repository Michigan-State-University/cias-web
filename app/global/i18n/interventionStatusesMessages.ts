import { defineMessages } from 'react-intl';

export const scope = 'app.global.InterventionStatusesMessages';

export default defineMessages({
  draft: {
    id: `${scope}.draft`,
    defaultMessage: 'Draft',
  },
  published: {
    id: `${scope}.published`,
    defaultMessage: 'Published',
  },
  closed: {
    id: `${scope}.closed`,
    defaultMessage: 'Closed',
  },
  archived: {
    id: `${scope}.archived`,
    defaultMessage: 'Archived',
  },
});
