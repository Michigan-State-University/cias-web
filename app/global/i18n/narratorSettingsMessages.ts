import { defineMessages } from 'react-intl';

import { NarratorSettingsKey } from 'models/Narrator';

export const scope = 'app.global.NarratorSettings';

export default defineMessages<NarratorSettingsKey>({
  [NarratorSettingsKey.VOICE]: {
    id: `${scope}.voice`,
    defaultMessage: 'Voice',
  },
  [NarratorSettingsKey.ANIMATION]: {
    id: `${scope}.animation`,
    defaultMessage: 'Display Narrator',
  },
  [NarratorSettingsKey.EXTRA_SPACE_FOR_NARRATOR]: {
    id: `${scope}.extra_space_for_narrator`,
    defaultMessage: 'Extra Space for Narrator',
  },
  [NarratorSettingsKey.CHARACTER]: {
    id: `${scope}.character`,
    defaultMessage: 'Character',
  },
});
