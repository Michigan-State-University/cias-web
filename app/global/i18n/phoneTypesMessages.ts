import { defineMessages } from 'react-intl';

import { PhoneType } from 'models/HfhsPatient';

export const scope = 'app.global.PhoneTypes';

export default defineMessages<PhoneType>({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  work: {
    id: `${scope}.work`,
    defaultMessage: 'Work',
  },
  mobile: {
    id: `${scope}.mobile`,
    defaultMessage: 'Mobile',
  },
});
