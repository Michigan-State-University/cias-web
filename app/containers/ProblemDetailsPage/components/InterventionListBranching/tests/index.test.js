/**
 *
 * Tests for InterventionListBranching
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { browserHistory } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import 'jest-styled-components';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'i18n';
import { Provider } from 'react-redux';
import configureStore from 'configureStore';

import InterventionListBranching from '../index';

const defaultProps = {
  nextInterventionName: 'test-2',
};

describe('<InterventionListBranching />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Provider store={store}>
          <InterventionListBranching {...defaultProps} />
        </Provider>
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Provider store={store}>
            <InterventionListBranching {...defaultProps} />
          </Provider>
        </IntlProvider>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
