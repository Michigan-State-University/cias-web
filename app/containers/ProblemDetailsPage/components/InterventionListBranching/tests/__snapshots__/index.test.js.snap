// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InterventionListBranching /> Should render and match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  justify-items: space-between;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  padding-top: 18px;
  padding-bottom: 18px;
  padding-right: 62px;
  padding-left: 62px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  justify-items: flex-end;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 25%;
}

.c8 {
  position: relative;
  display: inline-block;
  width: 40px;
  height: 21px;
  margin-left: 10px;
}

.c10 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  cursor: pointer;
  background-color: #CED3E4;
  border-radius: 21px;
}

.c9:checked + .c11 {
  background-color: #C866EA;
}

.c9:focus + .c11 {
  box-shadow: 0 0 1px #C866EA;
}

.c11:before {
  position: absolute;
  content: '';
  left: 4px;
  bottom: 4px;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  height: 13px;
  width: 13px;
  background-color: white;
  border-radius: 50%;
}

.c9:checked + .c11:before {
  -webkit-transform: translateX(18px);
  -ms-transform: translateX(18px);
  transform: translateX(18px);
}

.c3 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  font-size: 13px;
}

.c4 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-left: 10px;
  font-size: 13px;
  font-weight: 700;
}

.c7 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      width="100%"
    >
      <p
        className="c3"
        fontSize={13}
      >
        Next Session:
      </p>
      <p
        className="c4"
        fontSize={13}
        fontWeight="bold"
      >
        test-2
      </p>
    </div>
  </div>
  <div
    className="c5"
  >
    <div
      className="c6"
      width="100%"
    >
      <p
        className="c7"
      >
        Use formula
      </p>
      <label
        checked={false}
        className="c8"
      >
        <input
          checked={false}
          className="c9 c10"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c11"
        />
      </label>
    </div>
  </div>
</div>
`;
